### PR TITLE
exhaust: better build dir extraction and added user selection options

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR enables the user to select family/device/package/board when running exhaust.

### CODE REVIEW

File: exhaust.py [hunk 1]
```diff
@@ -150,7 +161,8 @@ def main():
         description='Exhaustively try project-toolchain combinations'
     )
     parser.add_argument('--family', default=None, help='device family')
-    parser.add_argument('--part', default=None, help='device part')
+    parser.add_argument('--device', default=None, help='FPGA device')
```
Comment (mithro) [inline exhaust.py:164]: Please add some examples to the help.

**Comment by mithro** [comment]
Merge if @kgugala approves.

File: exhaust.py (hunk 1 above)
Comment (acomodi) [reply exhaust.py]: Done

**Comment by acomodi** [comment]
There is a problem that, in CI, the result table that is printed does not contain all the tests that have been run.

Could be to the fact that all of them do fail.

@mithro Is Vivado available in CI, if so, the path is always the same? (e.g. `/opt/Xilinx/Vivado/...`)

**Comment by mithro** [comment]
@acomodi - I think it's reasonable to expect Vivado to always be installed in the same place.

**Comment by acomodi** [comment]
Green CI, merging.

**Pull request merged**